src/sentry/search/events/builder/discover.py

```diff
@@ -583,6 +584,10 @@ def resolve_params(self) -> list[WhereType]:
         if self.end:
             conditions.append(Condition(self.column("timestamp"), Op.LT, self.end))
 
+        # This will prevent calling Snuba with an empty list of projects, thus, returning no data.
+        if not self.params.project_ids:
+            raise UnqualifiedQueryError("You need to specify at least one project.")
```

Comment:
@wedamija @Zylphrex I'm now raising an error instead of relying on Snuba telling us.

I expect some backends to fail and I will adjust them.